src/OpenTelemetry.Collector.Dependencies/AzureSdkDiagnosticListener.cs

```diff
@@ -0,0 +1,86 @@
+﻿// <copyright file="AzureSdkDiagnosticListener.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Collector.Dependencies
+{
+    using System;
+    using System.Diagnostics;
+    using OpenTelemetry.Collector.Dependencies.Common;
+    using OpenTelemetry.Trace;
+
+    internal class AzureSdkDiagnosticListener : ListenerHandler
+    {
+        private readonly ITracer tracer;
+
+        private readonly ISampler sampler;
+
+        public AzureSdkDiagnosticListener(string sourceName, ITracer tracer, ISampler sampler)
+            : base(sourceName, tracer, null)
+        {
+            this.tracer = tracer;
+            this.sampler = sampler;
+        }
+
+        public void OnCompleted()
+        {
+        }
+
+        public void OnError(Exception error)
+        {
+        }
+
+        public override void OnStartActivity(Activity current, object valueValue)
+        {
+            var operationName = current.OperationName;
+            foreach (var keyValuePair in current.Tags)
+            {
+                if (keyValuePair.Key == "http.url")
+                {
+                    operationName = keyValuePair.Value;
+                    break;
+                }
+            }
+
+            var span = this.tracer.SpanBuilder(operationName)
+                .SetCreateChild(false)
+                .SetSpanKind(SpanKind.Client)
```

Comment:
oh, by the way, this is not always right
- http spans should have kind = client
- convenience layer spans should have kind = internal 

This is important for Azure Monitor and maybe other backends for better user experience.